worker.go

```diff
@@ -45,8 +45,17 @@ type Worker struct {
 
 func setSkipTLS(o *mqtt.ClientOptions) {
 	oldTLSCfg := o.TLSConfig
```

Comment:
Why not just set oldTLSCfg to the empty config, if o.TLSConfig is nil?

Reply:
Because the function should work in either case ... with an existing TLSConfig that the mqtt.ClientOptions contains or without (in case no other TLS options are set and the user only wants to not care about the server cert)

Reply:
Yeah,we both want to achieve the same effect. I would write it like this:
```
oldTLSCfg := o.TLSConfig
if oldTLSCfg == nil {
   oldTLSCfg = &tls.Config{}
}
oldTLSCfg.InsecureSkipVerify = true
o.SetTLSConfig(oldTLSCfg)
```

This code is IMHO easier to reason about because the if clause only handles the handling of the nil config. The rest of the code is for both cases the same. What is more, the is easier to change if we have to to more things with oldTLSCfg.

Reply:
@hikhvar  I agree, that looks much cleaner. I changed the PR accordingly.